app/serializers/serialize_solution_for_cli.rb

```diff
@@ -41,12 +41,11 @@ def user_handle
   def solution_url
     # TODO: Don't let someone download a personal_uuid and
     # end up with the mentor_uuid here. Actively guard this.
-    #
-    # TODO: How is this actually used within the CLI and could
-    # we just have it as nil if the user isn't the requestor?
-    return nil unless solution.user == requester
 
-    Exercism::Routes.private_solution_url(solution)
+    return Exercism::Routes.private_solution_url(solution) if solution.user == requester
+    return Exercism::Routes.mentoring_discussion_url(discussion) if requester.mentor? && discussion
+    return Exercism::Routes.published_solution_url(solution) if solution.published?
+    return Exercism::Routes.published_solution_url(solution) if requester.mentor? && mentor_request_pending?
```

Comment:
I've ordered them like this so save the query used in `mentor_request_pending`. I _could_ do the same for the `discussion` query where I first check if someone is _not_ a mentor. I don't know if that is a common scenario though.

Reply:
Two thoughts:
1. I think we should probably add a guard at the start of this for `return unless solution.viewable_by?(xxx)` so that method is still the central guard.
2. We should probably use the mentor request url for the final one, rather than the published solution one, which which case case we should probably put that first (although I appreciate the logic of what you've done!)